client/gutenberg/extensions/simple-payments/edit.js

```diff
@@ -3,34 +3,302 @@
+import { compose, withInstanceId } from '@wordpress/compose';
+import { Panel, PanelBody, PanelRow } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import { SIMPLE_PAYMENTS_PRODUCT_POST_TYPE } from 'lib/simple-payments/constants';
 
 class Edit extends Component {
-	handleChange = event => {
-		const paymentId = parseInt( event.target.value, 10 );
-		if ( paymentId ) {
-			this.props.setAttributes( { paymentId } );
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			savingProduct: false,
+		};
+	}
+
+	componentDidUpdate( prevProps ) {
+		const { simplePayment, attributes, setAttributes, isSelected, isSaving } = this.props;
+
+		if ( ! prevProps.simplePayment && simplePayment ) {
+			setAttributes( {
+				description: get( simplePayment, 'content.raw', attributes.description ),
+				currency: get( simplePayment, 'meta.spay_currency', attributes.currency ),
+				email: get( simplePayment, 'meta.spay_email', attributes.email ),
+				formattedPrice: get(
+					simplePayment,
+					'meta.spay_formatted_price',
+					attributes.formattedPrice
+				),
+				multiple: get( simplePayment, 'meta.spay_multiple', attributes.multiple ),
+				price: get( simplePayment, 'meta.spay_price', attributes.price ),
+				title: get( simplePayment, 'title.raw', attributes.title ),
+			} );
+		}
+
+		// Saves on block-deselect and when editor is saving a post
+		if ( ( prevProps.isSelected && ! isSelected ) || ( prevProps.isSaving && ! isSaving ) ) {
+			this.savePayment();
+		}
+	}
+
+	attributesToPost = attributes => {
+		const { title, description, currency, price, email, multiple } = attributes;
+
+		return {
+			title,
+			status: 'publish',
+			content: description,
+			featured_media: 0,
+			meta: {
+				spay_currency: currency,
+				spay_email: email,
+				spay_formatted_price: this.formatPrice( price ),
+				spay_multiple: multiple,
+				spay_price: price,
+			},
+		};
+	};
+
+	savePayment = async () => {
+		if ( this.state.savingProduct ) {
+			return;
+		}
+
+		this.setState( { savingProduct: true } );
+
+		const { attributes, setAttributes } = this.props;
+		const { paymentId } = attributes;
+
+		// @TODO field validation
+
+		const path = `/wp/v2/${ SIMPLE_PAYMENTS_PRODUCT_POST_TYPE }/${ paymentId ? paymentId : '' }`;
+
+		try {
+			// @TODO: then/catch
+			const { id } = await apiFetch( {
+				path,
+				method: 'POST',
+				data: this.attributesToPost( attributes ),
+			} );
+
+			this.setState( { savingProduct: false } );
+
+			if ( id ) {
+				setAttributes( { paymentId: id } );
+			}
+		} catch ( err ) {
+			// @TODO: error handling
+			this.setState( { savingProduct: false } );
+		}
+	};
+
+	// @FIXME: toFixed should be replaced with proper decimal calculations. See simple-payments/form
+	// const { precision } = getCurrencyDefaults( values.currency );
+	formatPrice = price => {
+		price = parseInt( price, 10 );
```

Comment:
Noting that it's odd to cut off the decimal part of the number when formatting. This means we're supporting only integers for prices, and I'd expect users would sometimes want non-integer prices - considering $x.99 is a price that we often see around.

Reply:
Agreed, I'll have improvements coming in the UI PR.